v2/tools/generator/internal/astmodel/internal_type_name_set.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"fmt"
+)
+
+// InternalTypeNameSet stores type names in no particular order without
+// duplicates.
+type InternalTypeNameSet map[InternalTypeName]struct{}
```

Comment:
Is there value in this (and the current `type_name_set.go` over and aobve what we already have in `v2/internal/set/set.go`? Wondering if we can just do do away with all of them in favor of the generic one?

Reply:
I started doing that and the scope of changes grew a bit too much for a single PR. It's a good idea, and one we should pursue as a follow-up.